src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -0,0 +1,43 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="GetDailyBuildNumber" 
+    AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"
+    Condition="Exists('$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll')" />
+
+  <Target Name="GetDailyBuildNumber"
+    BeforeTargets="Build"
+    Condition="'$(DailyBuildNumGenEnabled)' == 'true'"
+    >
+    <GetDailyBuildNumber
+      VersionPropsFile="$(SourceDir)version.props">
+      <Output PropertyName="DailyBuildNumber" TaskParameter="DailyBuildNumber" />
+    </GetDailyBuildNumber>
+  </Target>
+
+  <Target Name="CommitDailyBuildNumber"
+    AfterTargets="BuildPackages"
+    Condition="'$(DailyBuildNumGenEnabled)' == 'true'"
+    >
```

Comment:
And to answer your question, yeah I envision that this would be set via the official build but as it is right now there's nothing to prevent anybody from setting it.  We might want to guard against such a case, while not hard to overcome it would prevent some from publishing a package by accident (I don't want to gate getting this checked in on this issue though).